docs/dev_guide/contents/map_rotate_custom.rst

```diff
@@ -0,0 +1,90 @@
+.. _map_rotate_custom:
+
+****************************
+Adding a new rotation method
+****************************
+
+If you want to add a new rotation method that can be used by `sunpy.map.GenericMap.rotate`, you can do the following:
+
+.. code-block:: python
+
+    from sunpy.image.transform import add_rotation_function
+
+    @add_rotation_function("my_rotate",
+                        handles_clipping=False, handles_image_nans=False, handles_nan_missing=False)
+    def _rotation_mine(image, matrix, shift, order, missing, clip):
+        # Rotation code goes here
+        return rotated_image
+
+Then you can do:
+
+.. code-block:: python
+
+    from sunpy.map import Map
+    from sunpy.data import sample
+
+    hmi_map = sunpy.map.Map(sample.HMI_LOS_IMAGE)
+
+    rot_map = hmi_map.rotate(order=3, recenter=True, method="my_rotate")
+
+There are several things you will want to be aware of when adding a new rotation method.
+
+* The `.add_rotation_function` means you can handle nans, clipping, and missing values in the image array
```

Comment:
"you can handle" here is ambiguous, does the function have to handle it or the decorator. The decorator isn't "me".

I think we need to describe the expected inputs and outputs of a rotation function first. Things like you shouldn't modify the input array, the function + decorator must handles NaNs, handle missing values, clipping etc. Then after the expectations of the function are setup then we can say "and there is this decorator which can do these things for you".

Basically I am saying drop the list and describe the function, then introduce the details of the decorator as an aid.

Reply:
How about I just replicate the very detailed doc string that Albert wrote?

Reply:
Done.